### PR TITLE
java.lang.IllegalStateException: Exception occured during topic details retrieval. name: test

### PR DESCRIPTION
**Pull request by FridMehdi**

### CODE REVIEW

File: src/main/java/nb/kafka/operator/KafkaUtilities.java [hunk 1]
```diff
@@ -228,16 +231,22 @@ private void addAcl(Collection<AclBinding> acls, Collection<AclBinding> effectiv
   }
 
   TopicDescription topicDescription(String topicName) {
+    count = 0;
```
Comment (nbogojevic) [inline src/main/java/nb/kafka/operator/KafkaUtilities.java]: Any particular reason why this is on object level and not local variable?

File: src/main/java/nb/kafka/operator/KafkaUtilities.java [hunk 2]
```diff
@@ -228,16 +231,22 @@ private void addAcl(Collection<AclBinding> acls, Collection<AclBinding> effectiv
...
     DescribeTopicsResult dt = adminClient.describeTopics(singleton(topicName));
-    try {
-      dt.all().get();
-      TopicDescription topicDescription = dt.values().get(topicName).get();
-      return topicDescription;
-    } catch (InterruptedException | ExecutionException  e) {
-      if (e.getCause() != null && e.getCause() instanceof UnknownTopicOrPartitionException) {
-        return null;
+    while(true){
```
Comment (nbogojevic) [inline src/main/java/nb/kafka/operator/KafkaUtilities.java]: Looks good, can you just follow coding convention: space before brackets, braces:
`while (true) {`
and also encapsulate all if - else blocks in braces like:
```
if (++count == maxTries) {
...
} else {
...
}
```

File: src/main/java/nb/kafka/operator/KafkaUtilities.java (hunk 1 above)
Comment (FridMehdi) [reply src/main/java/nb/kafka/operator/KafkaUtilities.java]: i think that could be used in all methodes so i declared the MaxTries global static and  i init count in each function that we should retry

Comment (nbogojevic) [reply src/main/java/nb/kafka/operator/KafkaUtilities.java]: I'd make each count local, as then JVM can optimize it easier.

File: src/main/java/nb/kafka/operator/KafkaUtilities.java [hunk 3]
```diff
@@ -65,6 +65,10 @@
 
   private AdminClient adminClient;
 
+  final int maxTries = 10;
```
Comment (nbogojevic) [inline src/main/java/nb/kafka/operator/KafkaUtilities.java]: I suggest to rename this `private final static int MAX_TRIES`  to follow naming convention of ALL_CAPS for final statics (see [here](https://github.com/nbogojevic/kafka-operator/blob/master/src/main/java/nb/kafka/operator/ConfigMapManager.java) for example)

File: src/main/java/nb/kafka/operator/KafkaUtilities.java (hunk 1 above)
Comment (FridMehdi) [reply src/main/java/nb/kafka/operator/KafkaUtilities.java]: ok the issue still exits  
image

Comment (FridMehdi) [reply src/main/java/nb/kafka/operator/KafkaUtilities.java]: The issue persist
image

Comment (nbogojevic) [reply src/main/java/nb/kafka/operator/KafkaUtilities.java]: Can you check what has been setup on kafka side? This message means that we were unable to authorize, and I guess by default ENABLE_ACL is set to false in operator so it doesn't send credentials, and seems kafka deployment requires it. Can you see if it is the case, and disable acls on kafka side?

Comment (FridMehdi) [reply src/main/java/nb/kafka/operator/KafkaUtilities.java]: ok i see if acl is enabled in the broker we get this issue  maybe we need to  log something  if this exception is occured  ?

**Pull request closed without merging**